tests/system/src/test/java/test/robot/javafx/stage/StageOwnershipTest.java

```diff
@@ -0,0 +1,438 @@
+                stage1::show,
+                stage2::show,
+                () -> stage0.setIconified(true),
+                () -> {
+                    assertTrue(stage0.isIconified());
+                    assertTrue(stage1.isIconified());
+                    assertTrue(stage2.isIconified());
+                    assertColorDoesNotEqual(COLOR1, stage0);
+                    assertColorDoesNotEqual(COLOR2, stage1);
+                    assertColorDoesNotEqual(COLOR3, stage2);
+                },
+                () -> stage2.setIconified(false),
+                () -> {
+                    assertColorEquals(COLOR1, stage0);
+                    assertColorEquals(COLOR2, stage1);
+                    assertColorEquals(COLOR3, stage2);
+                });
+    }
+
+    @ParameterizedTest(name = PARAMETERIZED_TEST_DISPLAY)
+    @EnumSource(names = {"UNDECORATED", "DECORATED"})
+    void testMultipleChildren(StageStyle style) {
+        Util.runAndWait(() -> {
+            stage0 = createStage(style, COLOR1, null, Modality.NONE, -1, -1);
+            stage1 = createStage(style, COLOR2, stage0, Modality.NONE, -1, -1);
+            stage2 = createStage(style, COLOR3, stage0, Modality.NONE, -1, -1);
+        });
+        Util.doTimeLine(WAIT_TIME,
+                () -> {
+                    stage0.show();
+                    stage1.show();
+                    stage2.show();
+                },
+                () -> {
+                    stage2.setY(stage0.getY());
+                    stage1.setX(stage0.getX() - 300);
+                    stage1.setY(stage0.getY());
+                    stage2.setX(stage0.getX() + 300);
+                },
+                () -> stage0.setIconified(true),
+                () -> {
+                    assertColorDoesNotEqual(COLOR1, stage0);
+                    assertColorDoesNotEqual(COLOR2, stage1);
+                    assertColorDoesNotEqual(COLOR3, stage2);
+                },
+                () -> stage0.setIconified(false),
+                () -> {
+                    assertColorEquals(COLOR1, stage0);
+                    assertColorEquals(COLOR2, stage1);
+                    assertColorEquals(COLOR3, stage2);
+                });
+    }
+
+    @ParameterizedTest(name = PARAMETERIZED_TEST_DISPLAY)
+    @EnumSource(names = {"UNDECORATED", "DECORATED"})
+    void testClosingAppModalShouldFocusParent(StageStyle style) {
+        Util.runAndWait(() -> {
+            stage0 = createStage(style, COLOR1, null, Modality.NONE, -1, -1);
+            stage1 = createStage(style, COLOR2, null, Modality.NONE, -1, -1);
+            stage2 = createStage(style, COLOR3, stage0, Modality.APPLICATION_MODAL, -1, -1);
+        });
+
+        Util.doTimeLine(WAIT_TIME,
+                stage1::show,
+                stage0::show,
+                () -> stage1.requestFocus(),
+                stage2::show,
+                stage2::close,
+                () -> assertTrue(stage1.isFocused()));
```

Comment:
This test looks incorrect. If closing an app modal window should focus the parent (and I think it should) then stage0 should be focused.

Reply:
But `stage1` is the owner of `stage2`

Reply:
When stage2 is created you pass in stage0 as the owner, not stage1.

Reply:
Sorry, my bad, you're right. I now hide in shame..

Reply:
Fixed it.